src/main/java/net/snowflake/client/core/auth/wif/AwsIdentityAttestationCreator.java

```
@@ -29,17 +31,17 @@ public AwsIdentityAttestationCreator(AwsAttestationService attestationService) {
   }
 
   @Override
-  public WorkloadIdentityAttestation createAttestation() {
+  public WorkloadIdentityAttestation createAttestation() throws SFException {
     logger.debug("Creating AWS identity attestation...");
+    attestationService.initializeSignerRegion();
     AWSCredentials awsCredentials = attestationService.getAWSCredentials();
     if (awsCredentials == null) {
-      logger.debug("No AWS credentials were found.");
-      return null;
+      throw new SFException(
+          ErrorCode.WORKLOAD_IDENTITY_FLOW_ERROR, "No AWS credentials were found");
     }
     String region = attestationService.getAWSRegion();
     if (region == null) {
-      logger.debug("No AWS region was found.");
-      return null;
+      throw new SFException(ErrorCode.WORKLOAD_IDENTITY_FLOW_ERROR, "No AWS region was found");
```

Based on your changes to `getAWSRegion`, the region will never be null because metadata server returns either region or error (at least that's how it's typed in JS sdk)

---

In Java its still possible to return null I think. The error is silently catched and not returned in SDK
```
@Override
    public String getRegion() {
        if (SDKGlobalConfiguration.isEc2MetadataDisabled()) {
            throw new AmazonClientException("AWS_EC2_METADATA_DISABLED is set to true, not loading region from EC2 Instance "
                                         + "Metadata service");
        }

        if (region == null) {
            synchronized (this) {
                if (region == null) {
                    this.region = tryDetectRegion();
                }
            }
        }
        return region;
    }

    private String tryDetectRegion() {
        try {
            return EC2MetadataUtils.getEC2InstanceRegion();
        } catch (AmazonClientException sce) {
            LOG.debug("Ignoring failure to retrieve the region: " + sce.getMessage());
            return null;
        }
    }
 ```